### PR TITLE
QUINN-18 - Provide SparkSession in a class

### PR DESCRIPTION
**Pull request by afranzi**
This PR aims to provide the SparkSession inside a class instead of having a default Spark Session instantiated in `quinn.spark`.

This change should allow developers the issue of instantiating an unhandled spark session by mistake.

Besides, this PR has a proposal to provide the SparkSession as a session fixture to be injected in the desired tests.

What do you think @MrPowers?

### CODE REVIEW

File: quinn/spark.py [hunk 1]
```diff
@@ -1,7 +1,60 @@
...
+        self.spark = self.set_up_spark(app_name, self.master, conf, extra_dependencies, extra_files)
+
+    @property
+    def master(self) -> str:
+        return os.getenv('SPARK_MASTER', STANDALONE)
+
+    @staticmethod
+    def set_up_spark(app_name: str,
+                     master: str = STANDALONE,
+                     conf: SparkConf = None,
```
Comment (rberenguel) [inline quinn/spark.py]: I suspect mypy would complain here, should be `Optional[SparkConf]`

**Comment by MrPowers** [comment]
@afranzi - thanks for raising this PR.  I am trying to remember why I even have a SparkSession being created in the application code of this project in the first place...

spark-daria doesn't create a SparkSession for the application code - it [only creates a SparkSession for the test suite](https://github.com/MrPowers/spark-daria/blob/master/src/test/scala/com/github/mrpowers/spark/daria/sql/SparkSessionTestWrapper.scala).

Should we just move all of this code to the `/tests/` directory?

**Comment by afranzi** [comment]
In our case, we are using this `SparkProvider` to be inherited in our Spark jobs, so each job can overwrite its dependencies and extra py files to be added in their Spark Context.

We could add it in the core as a helper and example of how to initialize the SparkSession.
Otherwise, we can move it to test yes :)

**Comment by MrPowers** [comment]
@afranzi - I honestly don't know the best way to proceed here...

I suck at Python unfortunately, haha.

I like what you've outlined in this PR, especially if it sets a good example for how the PySpark community should manage the SparkSession in their projects.  So I'll follow your lead here, let me know if you still think this PR is how we should move this forward and I'll merge it in.

**Comment by afranzi** [comment]
I would keep it in the core.
@rberenguel what do you think about it? Should we provide it as a good practice to follow?

btw: I've fixed Travis by defining the missing library and the local setup (`pip install -e .` )

**Comment by rberenguel** [comment]
@afranzi it definitely looks like a good practice to suggest from a Python point of view. Modularisation is good

**Comment by MrPowers** [comment]
@afranzi @rberenguel - thanks for the help!

**Pull request merged**